File: backend/middleware/errorHandler.js
const errorHandler = (err,req,res,next)=>{
const statusCode = res.statusCode ? res.statusCode : 500;

const {constants} = require('../middleware/constants')

    switch(statusCode){
        case constants.VALIDATION_ERROR: 
            res.json({
                title: "Validation Failed - custom ",
                message: err.message,
                stacktrace : err.stacktrace
            })
            break;
        case constants.NOT_FOUND:
            res.json({
                title: "Not Found - custom ",
                message: err.message,
                stackTrace: err.stackTrace,
            });

        case constants.FORBIDDEN:
            res.json({
                title: "forbidden - custom",
                message: err.message,
                stacktrace: err.stackTrace
            })

        case constants.SERVER_ERROR:
            res.json({
                    title: "server error - custom",
                    message: err.message,
                    stacktrace: err.stackTrace
                })
        
        default:
            console.log("No error , all work ")
            break;
    }
    // res.json({message: err.message, stackTrace : err.stackTrace})

};


module.exports = errorHandler;

File: backend-pe/middleware/errorHandler.py
from fastapi import Request, HTTPException
from fastapi.responses import JSONResponse
from .constants import VALIDATION_ERROR, UNAUTHORIZED, FORBIDDEN, NOT_FOUND, SERVER_ERROR

async def error_handler(request: Request, call_next):
    try:
        response = await call_next(request)
        return response

    except Exception as e:
        return JSONResponse(content={"title": "Internal Server Error", "message": str(e)}, status_code=SERVER_ERROR)